### PR TITLE
fix:tests

### PR DESCRIPTION
**Pull request by livingrockrises**
revamp (WIP) test cases and helpers a bit to avoid running into stack too deep issues. 

for further work on this repo,
keep these in mind

i. Bundling related parameters into struct - those are used repeatedly.
ii. Minimize number of local variables and identify areas where it can be block scoped.
iii. Offload intermediate variables into memory to avoid direct stack operations
iv. Review order of variable initialisation
v. can work on further breaking down complex functions.

Note that paymaster validation gas limit, callGasLimit, verificationGasLimit estimations are currently skipped. either ways they can not be accurate without using gas estimation package.
unaccountedGasCost should be kept appropriate value after testing, when there is premium to be charge unaccountedGas needs to be higher value (also depending on warm/cold slot of fee collector)

### CODE REVIEW

File: test/base/TestBase.sol [hunk 1]
```diff
@@ -108,68 +117,65 @@ abstract contract TestBase is CheatCodes, TestHelper, BaseEventsAndErrors {
...
-        (userOp.paymasterAndData,) = generateAndSignPaymasterData(
-            userOp, PAYMASTER_SIGNER, paymaster, 3e6, 8e3, DAPP_ACCOUNT.addr, validUntil, validAfter, priceMarkup
+        PaymasterData memory pmData = PaymasterData(
+            100_000,
+            uint128(postOpGasLimitOverride),
+            DAPP_ACCOUNT.addr,
+            uint48(block.timestamp + 1 days),
+            uint48(block.timestamp),
+            priceMarkup
         );
```
Comment (filmakarov) [inline test/base/TestBase.sol]:
```suggestion
        PaymasterData memory pmData = PaymasterData({
            validationGasLimit: 100_000,
            postOpGasLimit: uint128(postOpGasLimitOverride),
            paymasterId: DAPP_ACCOUNT.addr,
            validUntil: uint48(block.timestamp + 1 days),
            validAfter: uint48(block.timestamp),
            priceMarkup: priceMarkup
        });
```

File: test/unit/concrete/TestSponsorshipPaymaster.t.sol [hunk 2]
```diff
@@ -205,13 +205,16 @@ contract TestSponsorshipPaymasterWithPriceMarkup is TestBase {
...
-        uint32 priceMarkup = 1e6;
+
+        startPrank(PAYMASTER_OWNER.addr);
+        bicoPaymaster.setUnaccountedGas(18_700);
+        stopPrank();
 
         PackedUserOperation[] memory ops = new PackedUserOperation[](1);
-        (PackedUserOperation memory userOp, bytes32 userOpHash) = createUserOp(ALICE, bicoPaymaster, priceMarkup);
+        // price markup of 1e6
+        (PackedUserOperation memory userOp, bytes32 userOpHash) = createUserOp(ALICE, bicoPaymaster, 1e6, 55_000);
```
Comment (filmakarov) [inline test/unit/concrete/TestSponsorshipPaymaster.t.sol:217]:
```suggestion
        (PackedUserOperation memory userOp, bytes32 userOpHash) = createUserOp({
        sender: ALICE, 
        paymaster: bicoPaymaster,
        pariceMarkup: 1e6,
        postOpGasLimitOverride: 55_000
  });
```

File: test/unit/concrete/TestSponsorshipPaymaster.t.sol [hunk 3]
```diff
@@ -231,17 +236,20 @@ contract TestSponsorshipPaymasterWithPriceMarkup is TestBase {
...
-        uint32 priceMarkup = 1e6 + 1e5;
 
+        startPrank(PAYMASTER_OWNER.addr);
+        bicoPaymaster.setUnaccountedGas(35_000);
+        stopPrank();
+
+        // 10% priceMarkup on gas cost
         PackedUserOperation[] memory ops = new PackedUserOperation[](1);
-        (PackedUserOperation memory userOp, bytes32 userOpHash) = createUserOp(ALICE, bicoPaymaster, priceMarkup);
+        (PackedUserOperation memory userOp, bytes32 userOpHash) = createUserOp(ALICE, bicoPaymaster, 1_100_000, 100_000);
```
Comment (filmakarov) [inline test/unit/concrete/TestSponsorshipPaymaster.t.sol:252]: same as above, name params for more readability

File: test/unit/fuzz/TestFuzz_TestSponsorshipPaymaster.t.sol [hunk 4]
```diff
@@ -133,9 +142,9 @@ contract TestFuzz_SponsorshipPaymasterWithPriceMarkup is TestBase {
         view
     {
         PackedUserOperation memory userOp = buildUserOpWithCalldata(ALICE, "", address(VALIDATOR_MODULE));
-        (bytes memory paymasterAndData, bytes memory signature) = generateAndSignPaymasterData(
-            userOp, PAYMASTER_SIGNER, bicoPaymaster, 3e6, 3e6, paymasterId, validUntil, validAfter, priceMarkup
-        );
+        PaymasterData memory pmData = PaymasterData(3e6, 3e6, paymasterId, validUntil, validAfter, priceMarkup);
+        (bytes memory paymasterAndData, bytes memory signature) =
+            generateAndSignPaymasterData(userOp, PAYMASTER_SIGNER, bicoPaymaster, pmData);
```
Comment (filmakarov) [inline test/unit/fuzz/TestFuzz_TestSponsorshipPaymaster.t.sol]:
```suggestion
        PaymasterData memory pmData = PaymasterData({
        validationGasLimit: 3e6,
        postOpGasLimit: 3e6,
        paymasterId: paymasterId,
        validUntil: validUntil,
        validAfter: validAfter, 
        priceMarkup: priceMarkup
    });
        (bytes memory paymasterAndData, bytes memory signature) =
            generateAndSignPaymasterData(userOp, PAYMASTER_SIGNER, bicoPaymaster, pmData);
```

**Review by filmakarov — APPROVED** [review]
approving with minor suggestions

File: test/unit/concrete/TestSponsorshipPaymaster.t.sol (hunk 3 above)
Comment (livingrockrises) [reply test/unit/concrete/TestSponsorshipPaymaster.t.sol:252]: was avoiding local vars :/

**Pull request merged**